src/containers/UserEdit.jsx

```diff
@@ -23,40 +23,93 @@ class UserEdit extends React.Component {
   };
 
   async componentWillMount() {
-    await this.props.mutations.editUser(null)
+    if (!this.props.authType === 'login' && !this.props.authType === 'signup') {
+      await this.props.mutations.editUser(null)
+    }
   }
 
   async handleSave(formData) {
-    await this.props.mutations.editUser(formData)
-    if (this.props.onRequestClose) {
-      this.props.onRequestClose()
+    if (this.props.authType === 'login' || this.props.authType === 'signup') {
+      // log in or sign up
+      const allData = {
+        nextUrl: this.props.location.query.nextUrl,
+        ...formData
+      }
+      const res = await fetch('/login-callback', {
+        method: 'POST',
+        body: JSON.stringify(allData),
+        headers: { 'Content-Type': 'application/json' },
+        redirect: 'follow'
+      })
+      const { redirected, status, url } = res
+      if (redirected && status === 200) {
+        this.props.router.push(url)
+      } else if (!redirected && status === 401) {
+        throw new Error('Invalid username or password')
+      } else {
+        throw new Error()
+      }
+    } else if (!this.props.authType === 'login' && !this.props.authType === 'signup') {
```

Comment:
Some style suggestions on if....else block here.

Generally I prefer the shorter IF block first, so the if...else here should probably be flipped.  Second, is the `else if` necessary? -- i.e. can we just have 'else'?  So combined, I think I would restructure this as:
```
if (!this.props.authType) {
  ... // await this.props.mutations.editUser(formData)....
} else {
   const allData = {
   ....
}
```